tools/fetch_remote_content/src/main.rs

```diff
@@ -1,16 +1,45 @@
-use std::env;
-use std::io;
+use std::{
+    env, io, str,
+    sync::{Arc, Mutex},
+};
 extern crate readability;
+use poptea::{GeminiClient, NoTrustStore};
 use readability::extractor;
 
+fn get_http(url: &str) -> Result<String, Box<dyn std::error::Error>> {
+    Ok(extractor::scrape(url)?.text)
+}
+
+fn get_gemini(url: &str) -> Result<String, Box<dyn std::error::Error>> {
+    let gemini_response = poptea::TlsClient::new(Arc::new(Mutex::new(NoTrustStore::default())))
+        .get(&url)
+        .map_err(|e| fetch_remote_err(&e.to_string()))?;
+    Ok(gemini_response
+        .body
+        .map(String::from_utf8)
+        .ok_or("no content")?
```

Comment:
Btw `Box<String>` is valid `Box<dyn Error>` so you don't need the `fetch_remote` helper function

Reply:
Fixed with `.map_err(|e| e.to_string())`